### PR TITLE
Adding more buffer_get_* runtime functions

### PR DESCRIPTION
**Pull request by parkhc**
buffer_get_dev, buffer_get_host_dirty, buffer_get_dev_dirty

### CODE REVIEW

**Comment by abadams** [comment]
I think you also need to update IR.cpp

File: src/runtime/buffer_t.cpp [hunk 1]
```diff
@@ -24,7 +29,17 @@ __attribute__((always_inline, weak))
 int _halide_buffer_get_max(const buffer_t *buf, int d) {
     return buf->min[d] + buf->extent[d] - 1;
 }
-    
+
+__attribute__((always_inline, weak))
+bool _halide_buffer_get_host_dirty(buffer_t *buf) {
```
Comment (abadams) [inline src/runtime/buffer_t.cpp]: const buffer_t *

File: src/runtime/buffer_t.cpp [hunk 2]
```diff
@@ -24,7 +29,17 @@ __attribute__((always_inline, weak))
...
+    return buf->host_dirty;
+}
+
+__attribute__((always_inline, weak))
+bool _halide_buffer_get_dev_dirty(buffer_t *buf) {
```
Comment (abadams) [inline src/runtime/buffer_t.cpp]: const buffer_t *

**Comment by parkhc** [comment]
changed to const. thanks.

**Comment by abadams** [comment]
LGTM

**Pull request merged**